0017-chart-namespace-overrides/README.md

```diff
@@ -0,0 +1,440 @@
+
+### Option 3 (Zarf Variable Style)
+
+**Given** I have a Zarf Package with a chart named `my-chart` in a component named `my-component`
+**And** the chart's namespace contains the variable `MY_CHART_NAMESPACE`
+```yaml
+variables:
+  - name: MY_CHART_NAMESPACE
+    default: the-namepace
+...
+ charts:
+  - name: my-chart
+    namespace: "###ZARF_VAR_MY_CHART_NAMESPACE###"
+```
+**When** I deploy that package with a `zarf-config.yaml` like the below*:
+```yaml
+package:
+  deploy:
+    set:
+      MY_CHART_NAMESPACE: new-namespace
+```
+**Then** Zarf will change the chart's release namespace to `new-namespace`
+
+**Given** I have a Zarf Package with a chart named `my-chart` in a component named `my-component`
+**And** the chart's namespace contains the variable `MY_CHART_NAMESPACE`
+```yaml
+variables:
+  - name: MY_CHART_NAMESPACE
+    default: the-namepace
+...
+ charts:
+  - name: my-chart
+    namespace: "###ZARF_VAR_MY_CHART_NAMESPACE###"
+```
+**When** I deploy that package with a `--set` like the below:
+```bash
+zarf package deploy zarf-package-test.tar.zst --set MY_CHART_NAMESPACE=new-namespace
+```
+**Then** Zarf will change the chart's release namespace to `new-namespace`
+
+#### Reason for Rejection
+
+This is one of the more "Zarf-way" options however, this would be the only place in the Zarf package definition that we allowed the `###ZARF_VAR` syntax to exist and the ### syntax is not very conducive to being maintained in a Zarf package (i.e. it is  treated as a comment when unquoted adding complexity for users and/or when trying to parse Zarf packages since this cannot be pre-templated like other `###ZARF_PKG_TMPL`s).
+
+### Option 4 (Prefix/Suffix Style)
+
+**Given** I have a Zarf Package with a chart deploying to `my-namespace`
+**When** I deploy that package with a `zarf-config.yaml` like the below*:
+```yaml
+package:
+  deploy:
+    namespace-prefix: new-
+    namespace-suffix: -kitteh
+```
+**Then** Zarf will change the chart's release namespace to `new-my-namespace-kitteh`
+
+**Given** I have a Zarf Package with a chart deploying to `my-namespace`
+**When** I deploy that package with a `--set` like the below:
+```bash
+zarf package deploy zarf-package-test.tar.zst --namespace-prefix new- --namespace-suffix -kitteh
+```
+**Then** Zarf will change the chart's release namespace to `new-my-namespace-kitteh`
+
+#### Reason for Rejection
+
+This would allow for some customization but may not provide enough flexibility in some cases.  Some clusters for security reasons will only authorize the deploy user to access specific namespaces which may not line up with what was originally in the package.  It also does not allow namespaces to be easily present elsewhere such as in `actions`.
+
+### Option 5 (Package Namespace Style)
```

Comment:
After reading this proposal again this feels like it may be the ideal way. In the current proposal if a namespace is not dynamic and the package is deployed twice to different namespaces, wouldn't both packages any charts in untemplated namespaces? This feels like it could lead to messy situations. 

If multiple namespaces are not going to be handled  well anyway it feels like we should instead aim for the more intuitive option.

Reply:
I agree with Austin here, with multiple namespace (which I don't think we want to initially support) we'll go into some sort of prefix/suffix situation. Or we'll have to expose multiple flags for multiple namespaces. Either of these approach is tricky and probably not something we want to tackle at this point in time. Let's start simple, supporting single namespace packages, and we can gather more data and better understand the multiple-namespace use cases.